lib/core/timer.cpp

```diff
@@ -14,4 +14,11 @@ Duration Timer::measure_execution_time(const std::function<void(void)> task)
   return after - before;
 }
 
+Duration Timer::measure_lapsed_time(const TimePoint previous_timepoint)
+{
+  const TimePoint time_now = time_.now();
+  const Duration duration  = previous_timepoint - time_now;
+  return duration;
```

Comment:
is it more readable to split this versus `return previous_timepoint - time_.now();`?

Reply:
i could see an arguement either way tbh

Reply:
I think I did it this way to both make it more readable and make it clear that `time_now` was explicitly a `TimePoint`